pkg/platform/model/inventory.go

```diff
@@ -271,7 +282,24 @@ func filterPlatformIDs(hostPlatform, hostArch string, platformIDs []strfmt.UUID)
 		return fallback, nil
 	}
 
-	return nil, &ErrNoMatchingPlatform{hostPlatform, hostArch}
+	return nil, &ErrNoMatchingPlatform{hostPlatform, hostArch, libcVersion}
+}
+
+func fetchLibcVersion() string {
+	if runtime.GOOS != "linux" {
+		return ""
+	}
+
+	var libcVersion string
+
+	libcInfo, err := sysinfo.Libc()
+	if err != nil {
+		logging.Error("Failed to fetch libc info: %v", err)
+		return ""
+	}
+	libcVersion = libcInfo.Version()
```

Comment:
nit: I think you can remove the `var libcVersion string` above and use `:=` here.